src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -373,7 +373,9 @@ public function get_items( $request ) {
 		update_post_author_caches( $query_result );
 		update_post_parent_caches( $query_result );
 
-		if ( post_type_supports( $this->post_type, 'thumbnail' ) ) {
+		$fields = $this->get_fields_for_response( $request );
+
+		if ( rest_is_field_included( 'featured_media', $fields ) || rest_is_field_included( '_links', $fields ) || rest_is_field_included( '_embedded', $fields ) ) {
```

Comment:
I'd think we'd still need the `post_type_supports` check, too, wouldn't we? There's not a benefit to updating the thumbnail cache if the post type doesn't support them.

I'm also wondering about the blanket checks for `_links` and `_embedded`. Should those be more targeted to `_links.wp:featuredmedia` e.g.

It also raises whether it'd be helpful to have a `rest_are_fields_included( string[] $field_names, $fields )` function

Reply:
We don't need `post_type_supports`. If the post type does not support featured images, then this field is missing.

Reply:
@spacedmonkey "missing" how? `rest_is_field_included` has no knowledge of the post type, so it only considers whether a parameter has been provided. And if no `_fields` is provided, it returns `true`. That means we _would_ end up calling `update_post_thumbnail_cache` for a post type which doesn't support them.

Reply:
@kadamwhite Featured media only in the schema if the post type supports it, then it would not be there. See
https://github.com/WordPress/wordpress-develop/blob/6222a118c9b87ca9f5703b5055831457469627b7/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L2488-L2494

Reply:
@spacedmonkey Ahh right, `rest_is_field_included` may not have any schema context but `get_fields_for_response` does build the included field list based on the specified schema. Good point! Concerns retracted.

Reply:
So 

```
if ( ( ( rest_is_field_included( '_embedded', $fields ) || rest_is_field_included( '_links', $fields ) ) && post_type_supports( $this->post_type, 'thumbnail' ) ) || rest_is_field_included( 'featured_media', $fields ) ) {
```